File: Mushka.WebApi/ClientModels/Product/ProductModel.cs
using System;

namespace Mushka.WebApi.ClientModels.Product
{
    public class ProductModel
    {
        public Guid Id { get; set; }

        public string Name { get; set; }

        public string VendorCode { get; set; }

        public decimal? RecommendedPrice { get; set; }

        public DateTime CreatedOn { get; set; }
        
        public Guid CategoryId { get; set; }

        public CategoryModel Category { get; set; }

        public SizeModel Size { get; set; }

        public bool IsArchival { get; set; }
    }
}

File: frontend2/src/app/shared/models/product.model.ts
import { Category } from './category.model';
import { Size } from './size.model';

export class Product {
  id: string;
  name: string;
  vendorCode: string;
  recommendedPrice: number;
  category: Category;
  categoryId: string;
  createdOn: string;
  size: Size;
  isArchival: boolean;
  subProducts: SubProduct[];

  constructor(data: any) {
    Object.assign(this, data);
  }
}

export class SubProduct {
  productId: string;
  quantity: number;

  constructor(data: any) {
    Object.assign(this, data);
  }
}
